### PR TITLE
RelNotes: Clone sig-release and use maps for website PR

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

During the last release cycle, we implemented the release notes maps feature to be able to edit release notes without changing the original pull requests. Changes using this feature were applied to the markdown draft but until now, the changes where not reflected in relnotes.k8s.io

This PR implements a change to the `krel release-notes` subcommand to clone k/sig-release and read the mapping files when generating the website PR. 

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

An example pull request generated with this change incorporating @celestehorgan  's changes into the v1.20-alpha.1 notes, can be inspected here:

https://github.com/kubernetes-sigs/release-notes/pull/203

#### Does this PR introduce a user-facing change?
```release-note
Pull requests done to the Release Notes website will now include edits from the release team. This are read from the mapping files stored in `kubernetes/sig-release`.
```

### CODE REVIEW

File: cmd/krel/cmd/release_notes.go [hunk 1]
```diff
@@ -787,6 +787,17 @@ func releaseNotesJSON(tag string) (string, error) {
...
 
+	logrus.Info("Cloning kubernetes/sig-release to read mapping files")
+	sigReleaseRepo, err := git.CleanCloneGitHubRepo(
+		git.DefaultGithubOrg, git.DefaultGithubReleaseRepo, true,
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "performing clone of k/sig-release")
+	}
+	defer func() {
+		err = sigReleaseRepo.Cleanup()
```
Comment (hasheddan) [inline cmd/krel/cmd/release_notes.go:798]: nitpick: I don't think assigning `err` here will have any effect unless you have a named return value for the encompassing function

**Review by hasheddan — COMMENTED** [review]
This is really cool! One nitpick below :)

File: cmd/krel/cmd/release_notes.go (hunk 1 above)
Comment (puerco) [reply cmd/krel/cmd/release_notes.go:799]: Good catch @hasheddan ! I found two like that in the file. They are fixed now.
Thanks !

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**